interactor/src/complete_flows/complete_flows_interactor_main.rs

```diff
@@ -250,16 +251,16 @@ impl CompleteFlowInteract {
         self.register_operation(shard, &hash_of_hashes, operations_hashes)
             .await;
 
-        self.register_token(shard, operation, None)
-            .await
-            .expect(FAILED_TO_REGISTER_SOVEREIGN_TOKEN)
+        self.register_token(shard, operation).await
     }
 
     pub async fn register_and_execute_sovereign_token(
         &mut self,
         mut config: ActionConfig,
         token: EsdtTokenInfo,
     ) -> EsdtTokenInfo {
+        let expected_deposit_log =
+            vec![log!(MULTI_ESDT_NFT_TRANSFER_EVENT, topics: [EGLD_000000_TOKEN_IDENTIFIER])];
```

Comment:
Here should be `build_expected_deposit_log` with additional `MULTI_ESDT_NFT_TRANSFER_EVENT`